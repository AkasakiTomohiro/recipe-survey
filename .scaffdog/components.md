---
name: 'components'
root: 'src/components'
output: '**/*'
ignore: ['__snapshots__']
questions:
  storyName: 'Story name. (e.g. Button, Header)'
  subLayer:
    message: 'Generate Sub Layer. (e.g. Header, XXX/YYY)'
    initial: '/'
---

# Variables

- Component: `{{ inputs.storyName | pascal }}`
- SubLayer: `{{ inputs.subLayer | pathConvert }}{{ Component }}/`

# `{{ SubLayer }}index.ts`

```typescript
export * from "./types";
export * from "./{{ Component }}";
```


# `{{ SubLayer }}{{ Component }}.stories.tsx`

```typescript
import { {{ Component }} } from './{{ Component }}';

import type { Meta, StoryObj } from '@storybook/react';

const meta: Meta<typeof {{ Component }}> = {
  title : 'RecipeSurvey/components/{{ SubLayer }}{{ Component }}',
  component: {{ Component }}
};

export default meta;

type Story = StoryObj<typeof {{ Component }}>;

export const Primary: Story = {
  args: {}
};

```

# `{{ SubLayer }}types/index.ts`

```typescript
export interface I{{ Component }}Props {
}
```

# `{{ SubLayer }}{{ Component }}.tsx`

```typescript

import { I{{ Component }}Props } from "./types";

/**
 * Primary UI component for user interaction
 */
export const {{ Component }} = (props: I{{ Component }}Props): JSX.Element => {

  return (
    <div>

    </div>
  );
};

```
