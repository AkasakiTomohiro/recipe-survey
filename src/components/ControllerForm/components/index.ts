/**
 * @file index.ts
 * @author Artan's Projects
 * @copyright Â© 2024 Artan's Projects. All rights reserved.
 */

export * from "./ControllerCheckboxForm";
export * from "./ControllerSelectForm";
export * from "./ControllerTextForm";
export * from "./GridFormControl";
export * from "./ListForm";