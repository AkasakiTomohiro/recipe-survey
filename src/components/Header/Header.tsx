
import * as React from 'react';

import { HelpCenter, Language } from '@mui/icons-material';
import DensityMediumOutlinedIcon from '@mui/icons-material/DensityMediumOutlined';
import { Button, FormControl, InputLabel, MenuItem, Select, SelectChangeEvent } from '@mui/material';

export const Header = (): JSX.Element => {

  const [language, setLanguage] = React.useState<string>('');
  const [game, setGame] = React.useState<string>('Satisfactory');

  const languageChange = (event: SelectChangeEvent) => {
    setLanguage(event.target.value)
  }
  const gameChange = (event: SelectChangeEvent) => {
    setGame(event.target.value);
  }

  return (
    <>
      <div style={{background: "#4CAF50", display: 'flex', alignItems: 'center', padding: '0 4px', width: 'full'}}>
        <Button variant='outlined'>
          <DensityMediumOutlinedIcon style={{color: 'black'}}/>
        </Button>
        <div className='roboto-bold' style={{paddingLeft: '8px'}}>
          Recipe Survey
        </div>
        <div style={{flexGrow: 1}}></div>
        <FormControl variant='standard' sx={{minWidth: '220px', m: '0 0px'}}>
          <InputLabel id='languageSelectLabel'>Language</InputLabel>
          <Select
            labelId='languageSelectLabel'
            id='languageSelect'
            value={language}
            onChange={languageChange}
            label='Language'
          >
            <MenuItem value={0}>
              English
            </MenuItem>
            <MenuItem value={10}>
              Japanese
            </MenuItem>
          </Select>
        </FormControl>
        <FormControl variant='standard' sx={{minWidth: '220px', m: '0 0px'}}>
          <InputLabel id='gameSelectLabel'></InputLabel>
          <Select
            labelId='gameSelectLabel'
            id='gameSelect'
            value={game}
            onChange={gameChange}
            label='Satisfactory'
          >
            <MenuItem value={0}>
              Satisfactory
            </MenuItem>
          </Select>
        </FormControl>
      </div>
    </>
  );
};
