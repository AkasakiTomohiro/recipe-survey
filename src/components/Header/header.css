@import url('https://fonts.googleapis.com/css2?family=Roboto:ital,wght@0,100;0,300;0,400;0,500;0,700;0,900;1,100;1,300;1,400;1,500;1,700;1,900&display=swap');

.roboto-bold {
  font-weight: 500;
  font-family: "Roboto", sans-serif;
  font-style: normal;
}

#header {
  background: #4CAF50;
  display: flex;
  alignItems: center;
  padding: 0 4px;
  width: full;
}

.select {
  min-width: 220px;
  margin: 0 10px;
}