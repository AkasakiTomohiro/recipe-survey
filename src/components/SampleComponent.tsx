import { useTransLocations } from '@/i18n';

export const SampleComponent = (): JSX.Element => {
  const t = useTransLocations();
  return (
    <>
      {t('docs')}
      {' '}
    </>
  );
};
