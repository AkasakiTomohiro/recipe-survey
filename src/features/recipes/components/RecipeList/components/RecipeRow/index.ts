/**
 * @file index.ts
 * @author Artan's Projects
 * @copyright Â© 2024 Artan's Projects. All rights reserved.
 */

export * from './RecipeRow';