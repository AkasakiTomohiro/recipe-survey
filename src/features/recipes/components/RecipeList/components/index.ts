/**
 * @file index.ts
 * @author Artan's Projects
 * @copyright Â© 2024 Artan's Projects. All rights reserved.
 */

export * from './SearchField';
export * from './RecipeRow';
