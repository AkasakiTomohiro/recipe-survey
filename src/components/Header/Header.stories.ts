/**
 * @file Header.stories.ts
 * @author Artan's Projects
 * @copyright Â© 2024 Artan's Projects. All rights reserved.
 */

import type { Meta, StoryObj } from '@storybook/react';
import { Header } from '~/components/Header';

// More on how to set up stories at: https://storybook.js.org/docs/writing-stories#default-export
const meta = {
  title     : './Header',
  component : Header,
  parameters: {
    // More on how to position stories at: https://storybook.js.org/docs/configure/story-layout
    layout: 'fullscreen',
  },

  // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/writing-docs/autodocs
  tags: ['autodocs'],

  // More on argTypes: https://storybook.js.org/docs/api/argtypes
  argTypes: {
    backgroundColor: { control: 'color' }
  }
} satisfies Meta<typeof Header>;

export default meta;
type Story = StoryObj<typeof meta>;

// More on writing stories with args: https://storybook.js.org/docs/writing-stories/args
export const Primary: Story = {
  args: {
  }
};