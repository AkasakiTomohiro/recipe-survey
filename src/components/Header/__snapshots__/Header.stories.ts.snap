// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`./Header Primary smoke-test 1`] = `
<div class="MuiBox-root css-12uhsvm">
  <header class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-t66lml-MuiPaper-root-MuiAppBar-root">
    <button class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-8gjog7-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
    >
      <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
           focusable="false"
           aria-hidden="true"
           viewbox="0 0 24 24"
           data-testid="DensityMediumOutlinedIcon"
           style="color: black;"
      >
        <path d="M3 3h18v2H3zm0 16h18v2H3zm0-8h18v2H3z">
        </path>
      </svg>
      <span class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root">
      </span>
    </button>
    <div class="MuiTypography-root MuiTypography-body1 css-1q8acoz-MuiTypography-root">
      Recipe Survey
    </div>
    <div class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary  css-z5brkq-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root">
      <div tabindex="0"
           role="combobox"
           aria-controls=":r0:"
           aria-expanded="false"
           aria-haspopup="listbox"
           aria-labelledby="languageSelectLabel languageSelect"
           id="languageSelect"
           class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
      >
        <span class="notranslate">
          ​
        </span>
      </div>
      <input aria-invalid="false"
             aria-hidden="true"
             tabindex="-1"
             class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
             value="日本語"
      >
      <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
           focusable="false"
           aria-hidden="true"
           viewbox="0 0 24 24"
           data-testid="ArrowDropDownIcon"
      >
        <path d="M7 10l5 5 5-5z">
        </path>
      </svg>
      <fieldset aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend class="css-yjsfm1">
          <span>
            Language
          </span>
        </legend>
      </fieldset>
    </div>
  </header>
</div>
`;
